### DISCUSSION TITLE
ysc to assembler. all about decompilation

### DISCUSSION THREAD

**Discussion by AIdaveloper** [YimMenu/YimMenu#1345] [General]
Hi all. This is a discussion about decompiling ysc game scripts to assembler and c.

In the beginning. I'm not good with assembler and new to decompiling and etc. I found an extension for Ghidra that allows you to decompile ysc but it doesn't work for me. Gives a memory error.
Ghidra ysc decompiler: [https://github.com/Parik27/ghidra_ysc](url)

So. I could not fix the error and therefore I am looking for a program to decompile ysc scripts.
And yes, I know about the ysc script decompiler. But it only decompiles to c. Whereas I also want to see assembler.
ysc script decompiler: [https://github.com/njames93/GTA-V-Script-Decompiler](url)

I want to collect here all the information about the decompilation of ysc scripts and find the best programs for this.

**Comment by tupoy-ya** [comment]
Probably the best decompiler for ysc scripts right now is [maybegreat48's](https://github.com/maybegreat48?tab=repositories) fork of [njames93's](https://github.com/njames93) decompiler:
https://github.com/maybegreat48/GTA-V-Script-Decompiler
But it can only disassemble single functions not entire scripts.

If you want a _probably_ working ghidra decompiler you can use [maybegreat48's](https://github.com/maybegreat48?tab=repositories) fork of [Parik27's](https://github.com/Parik27) decompiler:
https://github.com/maybegreat48/ghidra_ysc

**Reply by maybegreat48** [reply]
I'm pretty sure that the Ghidra decompiler doesn't work anymore

**Reply by AIdaveloper** [reply]
Thanks a lot. [maybegreat48's](https://github.com/maybegreat48?tab=repositories) fork works
